### PR TITLE
Update defaults of SignedXml to use SHA256

### PR DESCRIPTION
**Pull request by krwq**
SignedXml has previously used SHA1 by default. As SHA1 gets weaker, time has come that update of defaults is needed.

**This is a potentially breaking change.** If the other side of document exchange does not support SHA256 this will cause problems. Currently there is very little places which do not support it.
If you need to bring back the old behavior use:
```csharp
// SignatureMethod
SignedXml sxml = ...;
sxml.SignedInfo.SignatureMethod = SignedXml.XmlDsigRSASHA1Url;

// DigestMethod
Reference reference = ...;
reference.DigestMethod = SignedXml.XmlDsigSHA1Url;
...
```

FYI: @terrajobst @ericstj 

Fixes: https://github.com/dotnet/corefx/issues/18677

### CODE REVIEW

**Comment by tintoy** [comment]
Will this change be made in netfx, too, or only corefx?

**Comment by krwq** [comment]
@tintoy currently only here.
We do not have specific plans yet. Likely we will port this for netfx on the next release and will be quirked so it is opt-in when you target < next_version and enabled if you target >= next_version
@morganbr @bartonjs @AlexGhiondea - does that sound correct?

**Comment by krwq** [comment]
@tintoy most of the tests do not focus too much on the signature or digest method - also we do still want sha1 tests to work. All of the tests which had hard coded xml I've updated to still use SHA1 since that would be much more work without much value - not sure if this answers your question

File: src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public Reference()
             _refTarget = null;
             _refTargetType = ReferenceTargetType.UriReference;
             _cachedXml = null;
-            _digestMethod = SignedXml.XmlDsigSHA1Url;
+            _digestMethod = SignedXml.XmlDsigSHA256Url;
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs]: How about replacing this with an internal const for a default algorithm, so in the future replacement is easier?

File: src/System.Security.Cryptography.Xml/tests/SignedXmlTest.cs [hunk 2]
```diff
@@ -1543,5 +1550,51 @@ public void VerifyHMAC_HMACOutputLength_Invalid()
             SignedXml sign = GetSignedXml(xml);
             Assert.Throws<FormatException>(() => sign.CheckSignature(new HMACSHA1(Encoding.ASCII.GetBytes("no clue"))));
         }
+
+        [Fact]
+        public void SignedXmlUsesSha256ByDefault()
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Xml/tests/SignedXmlTest.cs]: Yay, you read my mind 😺.

**Comment by morganbr** [comment]
@krwq , that sounds plausible for netfx, but as you say, we don't have concrete plans yet.

File: src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs (hunk 1 above)
Comment (krwq) [inline src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/Reference.cs]: @bartonjs done

**Comment by anthonylangsworth** [comment]
Sorry for being a bit late. Still catching up after getting back from the Hawaii vacation. I think switching to SHA256 by default is a great idea.

**Pull request merged**